compiler/fir/fir2ir/src/org/jetbrains/kotlin/fir/backend/Fir2IrClassifierStorage.kt

```diff
@@ -282,11 +316,12 @@ class Fir2IrClassifierStorage(
     private fun createIrTypeParameterWithoutBounds(
         typeParameter: FirTypeParameter,
         index: Int,
-        typeContext: ConversionTypeContext = ConversionTypeContext.DEFAULT
+        typeContext: ConversionTypeContext = ConversionTypeContext.DEFAULT,
+        existingDescriptor: TypeParameterDescriptor? = null
     ): IrTypeParameter {
         require(index >= 0)
-        val descriptor = WrappedTypeParameterDescriptor()
         val origin = IrDeclarationOrigin.DEFINED
```

Comment:
For those new handled case(s), I wonder the origin of type parameter, which would be the same as that of the enclosing class, i.e., external stub, should be propagated and recorded in that way. How about moving this `origin` to the param of `createIrTypeParameterWithoutBounds` with default origin `DEFINED` and pass that external stub if necessary?

Reply:
done